### PR TITLE
HOTFIX aws-cn case for .ServiceAccountAnnotations for China regions

### PR DESCRIPTION
**Pull request by ced0ps**
This PR:

- adds/changes/removes etc

#### Optional app

- [ ] fresh install
- [ ] upgrade from previous version

#### Pre-installed app

- [ ] fresh install during cluster creation
- [ ] upgrade from previous version in a pre-existing cluster

#### Other testing

- [ ] check reconciliation of existing resources after upgrading

### Checklist

- [ ] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: helm/cert-manager-app/templates/_helpers.tpl [hunk 1]
```diff
@@ -118,6 +118,10 @@ Set Giant Swarm serviceAccountAnnotations.
 {{- define "giantswarm.serviceAccountAnnotations" -}}
 {{- if and (eq .Values.provider "aws") (eq .Values.controller.aws.irsa "true") (not (hasKey .Values.controller.serviceAccount.annotations "eks.amazonaws.com/role-arn")) }}
 {{- $_ := set .Values.controller.serviceAccount.annotations "eks.amazonaws.com/role-arn" (tpl "arn:aws:iam::{{ .Values.aws.accountID }}:role/{{ template \"aws.iam.role\" . }}" .) }}
+{{- else if and (eq .Values.provider "aws") (or (eq .Values.region "cn-north-1") (eq .Values.region "cn-northwest-1")) (eq .Values.aws.irsa "true") (not (hasKey .Values.controller.serviceAccount.annotations "eks.amazonaws.com/role-arn")) }}
+{{- $_ := set .Values.controller.serviceAccount.annotations "eks.amazonaws.com/role-arn" (tpl "arn:aws-cn:iam::{{ .Values.aws.accountID }}:role/{{ .Values.clusterID }}-Route53Manager-Role" .) }}
+{{- else if and (eq .Values.provider "aws") (eq .Values.aws.irsa "true") (not (hasKey .Values.controller.serviceAccount.annotations "eks.amazonaws.com/role-arn")) }}
```
Comment (Gacko) [inline helm/cert-manager-app/templates/_helpers.tpl:123]: Isn't this the same as in line 119?

Comment (Gacko) [reply helm/cert-manager-app/templates/_helpers.tpl:123]: With this #119 will always match and the CN exception will never be executed.

Comment (Gacko) [reply helm/cert-manager-app/templates/_helpers.tpl:123]: Ah, sorry. Missed the `controller` thingy.

**Review by Gacko — APPROVED** [review]

**Comment by ced0ps** [comment]
This is not implemented in the same way in China Clusters so this PR is not required.

**Pull request closed without merging**